packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -287,6 +316,17 @@ export class DebugConfigurationManager {
         return uri;
     }
 
+    private async getFolderConfigurationUri(model: DebugConfigurationModel): Promise<URI> {
+        const paths = this.preferenceConfigurations.getPaths();
+        for (const configPath of paths) {
+            const configUri = new URI(model.workspaceFolderUri).resolve(`${configPath}/launch.json`);
+            if (await this.filesystem.exists(configUri.toString())) {
```

Comment:
why do we need it? if something exist already we should not override it at all, previous fallback was fine to me

Reply:
that sounds like a bug in the preference service, `DebugConfigurationModel.uri` used to point to a config from which data was loaded or nothing

Reply:
@akosyakov 
Before my change we use `DebugConfigurationModel.uri`, which works perfectly in a workspace that has only one folder. 
In a multi root workspace. `DebugConfigurationModel.uri` points to the `theia-workspace` file - it is correct, however does not work well since we want the URI that points to the config file in the folder.